### ISSUE TITLE
Python hyperqueue package for ARM

### ISSUE THREAD

**Issue by theely** [It4innovations/hyperqueue#726]
Hi there,
Would it be possible to release the hyperqueue API client also for the ARM architecture?
Or alternatively, I am open to suggestions on how to enable the development that makes use of your API client on Apple silicon. 

Cheers,
Elia

**Comment by Kobzol** [comment]
Hi, just to clarify, you are looking for precompiled HyperQueue Python API package that would work on ARM (Apple Silicon)?

**Comment by theely** [comment]
Yes exactly, a package that can be installed via pip install ...

**Comment by Kobzol** [comment]
I see. So, currently we offer pre-built binary HyperQueue packages for x64, PowerPC64 and ARM64 Linux, and a pre-built Python HyperQueue package for x64 Linux.

We could in theory expand the number of packages that we offer, but a larger issue is that HyperQueue doesn't really support macOS (you can find more information here: https://github.com/It4innovations/hyperqueue/issues/688). While it can be currently compiled for it, we do not offer any support for it, and some functionality will simply not work on other operating systems than Linux, at the moment.

Since macOS is not supported, and the Python API itself is still very experimental, we probably will not support prebuilt macOS Python packages anytime soon. I'm not even sure how can we build such packages, but if you investigate and find some solution, I'd be glad if you can post it here.

**Comment by theely** [comment]
Thanks for the fast responses.
I followed your advice and built my own binary. I was able to compile the Python package on MacOS with no issues.
If it helps, I am happy to share the binary (hyperqueue-0.19.0-cp36-abi3-macosx_11_0_arm64.whl)?

**Comment by Kobzol** [comment]
I'm glad that it worked for you! Providing a single wheel is not enough to make this robust for us, though. We'd need a way to produce the wheels on our CI, using `maturin`. [This](https://github.com/It4innovations/hyperqueue/blob/main/.github/workflows/build.yml#L134) is the CI job that builds our Linux Python package. It would probably need to run in a macOS container so that it could generate a macOS ARM Python wheel.

**Comment by Kobzol** [comment]
Okay, if maturin supports it out of the box, then it shouldn't be that hard. I don't have bandwidth to work on this right now, but I'd happy to review a PR :)
